internal/bminventory/inventory.go

```diff
@@ -4079,6 +4093,35 @@ func (b *bareMetalInventory) validateClusterInfraEnvRegister(ctx context.Context
 	return nil
 }
 
+func (b *bareMetalInventory) setIgnitionConfigOverrideUsage(clusterId strfmt.UUID, ignitionConfigOverride, level string, db *gorm.DB) error {
+	var cluster *common.Cluster
+	cluster, err := common.GetClusterFromDBForUpdate(db, clusterId, common.SkipEagerLoading)
+	if err != nil {
+		return err
+	}
+	if usages, uerr := usage.Unmarshal(cluster.Cluster.FeatureUsage); uerr == nil {
+		props := usages[usage.IgnitionConfigOverrideUsage].Data
+		if props == nil {
+			props = make(map[string]interface{})
+		}
+		ignitionOverriden := ignitionConfigOverride != ""
+		props[level] = ignitionOverriden
+
+		if !ignitionOverriden {
+			// Check to see if at least one level of ignition config override is set
+			for _, val := range props {
+				if val.(bool) {
+					ignitionOverriden = true
+				}
+			}
+		}
```

Comment:
I modified it! It was supposed to determine whether the feature usage was actually used by iterating through all the levels since this function could be called for a level where the usage isn't set.

Example: if at the host level, the ignition override was set but this function was called from the infra-env level where the override wasn't set. We don't want to set the overall feature usage to false.

Does this seem like a reasonable check?